include/kafka/Error.h

```diff
@@ -62,5 +64,59 @@ struct ErrorWithDetail
     std::string     detail;
 };
 
+/**
+ * Error type with rich info.
+ */
+class Error
+{
+public:
+    explicit Error(rd_kafka_error_t* error = nullptr): _rkError(error, RkErrorDeleter) {}
+    Error(const Error&)            = default;
+    virtual ~Error()               = default;
+
+    /**
+     * Obtains the underlying error code.
+     */
+    virtual std::error_code     errorCode()   const { return ErrorCode(rd_kafka_error_code(_rkError.get())); }
+
+    /**
+     * Readable error string.
+     */
+    virtual std::string         message()     const { return rd_kafka_error_string(_rkError.get()); }
+
+    /**
+     * Fatal error indicates that the client instance is no longer usable.
+     */
+    virtual std::optional<bool> isFatal()     const { return rd_kafka_error_is_fatal(_rkError.get()); }
```

Comment:
Potentially, we can just return bool type instead of std::optional<bool>

Reply:
In fact, I don't like to use the `std::optional` either, -- while, if the `Error` is emitted from a `rd_kafka_resp_err_t`, there's no way to figure out whether it's `fatal` or not. I guess, eventually `librdkafka` would adapt all interfaces with `rd_kafka_error_t` (instead of `rd_kafka_resp_err_t`), then we could remove the `optional` here.